src/librustc/ty/instance.rs

```diff
@@ -80,7 +81,7 @@ impl<'tcx> Instance<'tcx> {
         // There shouldn't be any params - if there are, then
         // Instance.ty_env should have been used to provide the proper
         // ParamEnv
-        if self.substs.has_param_types() {
+        if self.substs.has_param_types() && !tcx.sess.opts.debugging_opts.polymorphize {
             bug!("Instance.ty called for type {:?} with params in substs: {:?}", ty, self.substs);
```

Comment:
I forget where/if I left a comment, but the current situation with `Instance`'s types and `Instance::ty_env` in particular is pretty bad and we should probably require passing `ParamEnv` around a lot (or maybe start using `ParamEnvAnd<Instance>`?).

Reply:
I think it was https://github.com/rust-lang/rust/pull/67800#discussion_r386685155.